src/ds/dl_list.h

```diff
@@ -6,6 +6,8 @@
 #include <cstdint>
 
 #define SNMALLOC_ASSERT assert
```

Comment:
I don't think we should have those, instead we should include the right PAL/AAL and get the right includes.